### DISCUSSION TITLE
Correct config setting for using skins with Gem file?

### DISCUSSION THREAD

**Discussion by rjp23** [mmistakes/minimal-mistakes#3267] [Q&A] [answered]
I know this is in the documentation but I'm not finding it as clear as I suspect I should be.

- I'm hosting via gh-pages.
- I forked the mm-github-pages-starter repo and am using the Gem file.
- I want to change the skin colour.

The instructions say:

> If you’re using the Ruby gem version of the theme you’ll need this line to activate it:
> > theme: minimal-mistakes-jekyll
> 
> Skin
> > Easily change the color scheme of the theme using one of the provided “skins”:
> > minimal_mistakes_skin: "default" # "air", "aqua", "contrast", "dark", "dirt", "neon", "mint", "plum" "sunrise"

I'm interpreting this as that my _config.yml file should contain:

theme                  : "minimal-mistakes-jekyll"
#remote_theme           : "mmistakes/minimal-mistakes"
minimal_mistakes_skin    : "air"

(i.e. I comment out remote theme and add "theme" in it's place).

That doesn't seem to work. I've tried all the combinations of those 3 options and I can't seem to get it to change the theme.

I'm also not clear if the skin value should be "air" or air. The guide says "default" (with quotes) but the config file omits the quote marks.

I'm sure I'm missing something simple... 

Thanks

**Comment by mmistakes** [comment, marked as answer]
There is almost no difference between using the theme as a Ruby gem (via the `theme` key) or as a remote theme (via the `remote_theme` key). The main difference is how the theme files are installed.

If you're hosting with GH Pages you need to use the `remote_theme` option. GitHub Pages does not let you install arbitrary 3rd party gems unless they're from a small list of approved gems.

You can set the skin with `air` or `"air"`. Both are strings as far as YAML is concerned. There are some special characters in YAML that can give you errors or have unintended uses, so placing your YAML strings in quotes is generally safer.

TL/DR: set your _config.yml to:

```
remote_theme: "mmistakes/minimal-mistakes"
minimal_mistakes_skin: "air"
```

If that's not working for you provide a link to your repo so I can inspect it. I just forked the [mm-github-pages-starter](https://github.com/mmistakes/mm-github-pages-starter) repo changed `minimal_mistakes_skin: default` to `minimal_mistakes_skin: air` and I'm seeing the appropriate skin when testing locally.

![Screen Shot 2021-12-04 at 7 57 16 PM](https://user-images.githubusercontent.com/1376749/144729287-d2b1331e-0932-4046-aaca-1d7b97bd71ee.png)

**Reply by beyond9thousand** [reply]
It doesn't work for me, the github remote theme method. am i doing something wrong?
https://github.com/beyond9thousand/techsupport

**Reply by barriejc** [reply]
Technically this selected answer works for me, but [the issue](https://github.com/mmistakes/minimal-mistakes/discussions/3267#discussioncomment-1752145) way down at the bottom reported by OP rjp23 is what I'm experiencing, but locally on Mac OSX rather than with GitHub pages / remote_theme.  The iBug response is not helpful or applicable, because it's more about any _config.yml change not taking effect when using --watch, whereas I am willing to restart Jekyll.

When I initially cloned the repo and changed the skin to neon, it worked.  Then I changed other things like title, email, website, GitHub, added a LinkedIn item, etc. and now the skin `will. not. change.` no matter how many times I ctrl-c and `jekyll serve`, or `jekyll build`.  Can't get it to go back to default or any other skin. Help?

**Reply by barriejc** [reply]
a ha! Moving the comment after "default" or whatever skin I configure to the next line did it.  Then I ctrl+c, `jekyll build` and `jekyll serve` (didn't try with --livereload or watch).

Before:
```
minimal_mistakes_skin  : "dark" # "air", "aqua", "contrast", "dark", "dirt", "neon", "mint", "plum", "sunrise"
```

After:
```
minimal_mistakes_skin  : "dark"
# "air", "aqua", "contrast", "dark", "dirt", "neon", "mint", "plum", "sunrise"
```

**Reply by rishabh96b** [reply]
My `_config.yaml` is set in accordance with above instructions but the theme (dark) which I am trying to set is not working.
Here is the link to my config: [rishabh96b.github.io/blob/main/_config.yml](https://github.com/rishabh96b/rishabh96b.github.io/blob/main/_config.yml)

**Comment by rjp23** [comment]
Thanks for the help!

I think my issue might be that github was taking a **long** time to build the page so I thought it wasn't working when it was.

**Comment by rjp23** [comment]
Hmm actually, what I said isn't quite true.

It seems to rebuild the content very quickly (e.g. if I change the title, it'll change almost instantly). However, even if I change the skin **at the same time** as changing the title, the title will change but the skin won't.

**Reply by iBug** [reply]
https://github.com/jekyll/jekyll/issues/2302

**Comment by dejang95** [comment]
If anyone stumbles upon this searching for an issue, make sure you refresh without cache. Solved it for me.

**Reply by frank20a** [reply]
Spent way too much time trying to fix this...
Thanks for the comment it finally worked!
